### PR TITLE
fix: use scratch buffer and lhs for keymap.normalize()

### PR DESCRIPTION
**Pull request by zeertzjq**
Fix https://github.com/neovim/neovim/issues/18303

This can be slower, I don't know how frequently this function is called as I don't use this plugin.

### CODE REVIEW

**Comment by zeertzjq** [comment]
@arsham Can you test if this fixes <https://github.com/neovim/neovim/issues/18303>?

File: lua/cmp/utils/keymap.lua [hunk 1]
```diff
@@ -16,12 +16,15 @@ end
...
 ---@return string
 keymap.normalize = function(keys)
-  vim.api.nvim_set_keymap('t', '<Plug>(cmp.utils.keymap.normalize)', keys, {})
-  for _, map in ipairs(vim.api.nvim_get_keymap('t')) do
-    if keymap.equals(map.lhs, '<Plug>(cmp.utils.keymap.normalize)') then
-      return map.rhs
+  local normalize_buf = vim.api.nvim_create_buf(false, true)
+  vim.api.nvim_buf_set_keymap(normalize_buf, 't', keys, '<Plug>(cmp.utils.keymap.normalize)', {})
+  for _, map in ipairs(vim.api.nvim_buf_get_keymap(normalize_buf, 't')) do
+    if keymap.equal(map.rhs, '<Plug>(cmp.utils.keymap.normalize)') then
```
Comment (arsham) [inline lua/cmp/utils/keymap.lua:22]: This should be `keymap.equals`.

**Comment by arsham** [comment]
> @arsham Can you test if this fixes [neovim/neovim#18303](https://github.com/neovim/neovim/issues/18303)?

It works, thank you!

File: lua/cmp/utils/keymap.lua (hunk 1 above)
Comment (zeertzjq) [reply lua/cmp/utils/keymap.lua:22]: Oops

**Comment by hrsh7th** [comment]
LGTM! It looks reasonable. Thank you!

**Pull request merged**